### PR TITLE
fix: jsx files are not taken into front-end build from add-ons

### PR DESCRIPTION
**Pull request by mstahv**
We already support tsx, but the more trivial jsx files don't seem to work. Something that people probably try first, especially if the embedded component is only using JS...

## Type of change

- [x] Bugfix
- [] Feature

## Checklist

- [ ] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [ ] I have added a description following the guideline.
- [ ] The issue is created in the corresponding repository and I have referenced it.
- [ ] I have added tests to ensure my change is effective and works as intended.
- [ ] New and existing tests are passing locally with my change.
- [ ] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

**Comment by mcollovati** [comment]
Thanks, @mstahv.
Do you mind updating the tests to cover the added extensions, as done in #13323?

**Comment by mstahv** [comment]
I was looking at that for a while, but modifying those jar files to include test files made me feel reluctant 😁

**Comment by mstahv** [comment]
Thanks @caalador, here is a medal for you 🥇

File: flow-server/src/test/java/com/vaadin/flow/server/frontend/TaskCopyFrontendFilesTest.java [hunk 1]
```diff
@@ -153,8 +153,15 @@ private void should_collectJsAndCssFilesFromJars(String jarFile,
                 files.contains("react.tsx"));
 
         Assert.assertTrue(
-                "TSX resource source map should have been copied from jar file",
+                "JSX resource source map should have been copied from jar file",
```
Comment (mcollovati) [inline flow-server/src/test/java/com/vaadin/flow/server/frontend/TaskCopyFrontendFilesTest.java:156]:
```suggestion
                "TSX resource source map should have been copied from jar file",
```

File: flow-server/src/test/java/com/vaadin/flow/server/frontend/TaskCopyFrontendFilesTest.java [hunk 2]
```diff
@@ -153,8 +153,15 @@ private void should_collectJsAndCssFilesFromJars(String jarFile,
...
                 files.contains("react.tsx.map"));
+
+        Assert.assertTrue("TSX resource should have been copied from jar file",
```
Comment (mcollovati) [inline flow-server/src/test/java/com/vaadin/flow/server/frontend/TaskCopyFrontendFilesTest.java:159]:
```suggestion
        Assert.assertTrue("JSX resource should have been copied from jar file",
```

**Review by mcollovati — CHANGES_REQUESTED** [review]
Just a typo in test error messages

**Review by mcollovati — APPROVED** [review]

**Pull request merged**